packages/filebrowser/src/browser.ts

```diff
@@ -249,6 +249,17 @@ export class FileBrowser extends SidePanel {
     return this.listing.paste();
   }
 
+  /**
+   * Dispose of the resources held by the browser.
+   */
+  dispose(): void {
+    if (this.isDisposed) {
+      return;
+    }
+    this.model.dispose();
```

Comment:
I see why you moved it here, though I am worried it could backfire because model is not created by the browser but given to it in options. I think we should use previous implementation as this may lead to disposal of model which is still used elsewhere.